src/main/java/com/aws/greengrass/detector/uploader/ConnectivityUpdater.java

```diff
@@ -79,10 +82,10 @@ synchronized void uploadAddresses(List<String> ips, Config config) {
     }
 
     //Default for JUnit Testing
-    boolean hasIpsChanged(@NonNull List<String> ips) {
+    boolean hasIpsOrPortChanged(@NonNull List<String> ips, int port) {
```

Comment:
Lets just keep a separate function for `hasPortChanged` and use `if (hasIpsChanged || hasPortChanged)`
Smaller functions with single responsibilities are preferred.

Reply:
Done